electrumsv/wallet.py

```diff
@@ -2768,6 +2768,22 @@ def update_cached_values(callback_future: concurrent.futures.Future[PasswordUpda
             keystore_by_masterkey_id = { keystore_id: keystore for keystore_id, keystore
                 in self._keystores.items() if keystore.can_change_password() }
 
+            # Update cached server `encrypted_api_key` values. This is especially important given
+            # that when the wallet shuts down, `update_network_servers_transaction` flushes any
+            # server updates which would revert the change in the database.
+            for server_row in self.data.read_network_servers():
```

Comment:
If you are updating the cached values, then is it safe to use the non-cached values?

Reply:
What are you concerned about? This update of the cached values only affects the `encrypted_api_key` field and occurs as a callback immediately after the database is updated to the new re-encrypted value (with the new password).

Reply:
Why don't you just update the cached values directly?

Reply:
Because wont that clobber the `date_last_try` & `date_last_good` when I only want to update the `encrypted_api_key` while leaving any other state unchanged.

Reply:
I do not understand why you are reading in the database rows here, they are completely irrelevant. Just update the cached credentials directly like all the other credentials. That is the purpose of this function.

Reply:
At first I didn't know what you meant - I am updating this part now...